tests/lms/views/error_test.py

```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+from pyramid import httpexceptions
+import pytest
+
+from lms.views import error
+
+
+class TestErrorView:
+    def test_it_sets_correct_message_for_http_error(self, pyramid_request):
+        exc = httpexceptions.HTTPError('test http error msg', status_code=500)
+
+        error_view = error.ErrorView(exc, pyramid_request)
+        resp = error_view.httperror()
+
+        assert resp['message'] == 'test http error msg'
+
+    def test_it_sets_correct_request_status_int_for_http_error(self, pyramid_request):
+        exc = httpexceptions.HTTPError('test http error msg', status_code=500)
+
+        error_view = error.ErrorView(exc, pyramid_request)
+        error_view.httperror()
+
+        assert pyramid_request.response.status_int == exc.status_int
+
+    def test_it_sets_correct_message_for_http_server_error(self, pyramid_request):
+        exc = httpexceptions.HTTPServerError('test server error msg', status_code=500)
+
+        error_view = error.ErrorView(exc, pyramid_request)
+        resp = error_view.httperror()
+
+        assert resp['message'] == 'test server error msg'
+
+    def test_it_sets_correct_request_status_int_for_http_server_error(self, pyramid_request):
+        exc = httpexceptions.HTTPServerError('test server error msg', status_code=500)
+
+        error_view = error.ErrorView(exc, pyramid_request)
+        error_view.httperror()
+
+        assert pyramid_request.response.status_int == exc.status_int
+
+    def test_it_sets_correct_message_for_non_http_error(self, pyramid_request):
+        exc = Exception()
+
+        error_view = error.ErrorView(exc, pyramid_request)
+        resp = error_view.error()
+
+        assert resp['message'] == 'Sorry, but something went wrong. The issue has been reported and we\'ll try to fix it.'
+
+    def test_it_logs_non_http_error_in_sentry(self, pyramid_request):
+        exc = Exception()
+
+        error_view = error.ErrorView(exc, pyramid_request)
+        error_view.error()
+
+        pyramid_request.raven.captureException.assert_called_once()
+
+    def test_it_re_raises_exception_in_debug_mode(self, pyramid_request):
+        exc = Exception()
```

Comment:
I think you should test with a more specific exception here because the `pytest.raises(Exception)` below will pass for _any_ exception raised, not just for when the given exception is re-raised. You can probably still use the `Exception` class but maybe pass it a string message and then use the `match` arg to `raises`